lib/audit_test.pm

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Base module for audit-test test cases
+# Maintainer: llzhao <llzhao@suse.com>
+# Tags: poo#
+
+package audit_test;
+
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Mojo::File 'path';
+
+use base 'consoletest';
+
+our @EXPORT = qw(
+  $testdir
+  $testfile_tar
+  $baseline_file
+  $code_repo
+  $mode
+  run_testcase
+  parse_testcase_log
+  compare_run_log
+);
+
+our $testdir      = "/tmp/";
+our $code_repo    = get_var("CODE_BASE", "https://gitlab.suse.de/security/audit-test-sle15/-/archive/master/audit-test-sle15-master.tar");
```

Comment:
This hardcodes "sle15" within the string and I don't see a good reason for that. Better avoid encoding versions anywhere except for cases like `is_sle('<15')` which are easier to manage for version ranges rather than pinning individual versions. I know that this relates to an external repository. Maybe it can be changed in there. If not, how about parsing the codestream from the "VERSION" field as you already parse the "testsuite_name" later in a similar way?

Also our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
The default `CODE_BASE` is owned by "Security Team" and "Cert Team". QE security team don't write permission atm.
But there is a settings `CODE_BASE` can chose the version if needed.

Yes, you are right and we all know/agree on "Factory First" policy.
But "audit-test" (not been updated for more than 3 years) is for CCC certification testing purpose only atm as it need lots of specific environments, e.g., FIPS setup. So we only test sles.

It might be considered to test TW in the future but atm the human resource is really limited.

Reply:
Well, human resources are always limited :) This is why I bring up my suggestions so that we can have something that is long-term maintainable. The question is only "do it now or do it later" where I suggest that the "do it later" decision is still taken too often :)

Reply:
We agree with and have a log-term plan for the Factory First.
It is the truth that we have other urgent tasks to do atm, such as done the short-term tasks on "CC" (common criteria) testings before 15sp4 which handed over by kernel team.